File: 167.two-sum-ii-input-array-is-sorted.py
#
# @lc app=leetcode id=167 lang=python3
#
# [167] Two Sum II - Input array is sorted
#

# @lc code=start
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        target_map={}
        for index,num in enumerate(numbers):
            if num in target_map.keys():
                break
            target_map[target-num]=index
        index1 = target_map[num]
        index2 = index
        if index1 > index2:
            index1,index2 = index2,index1
        return [index1+1,index2+1]


# @lc code=end

File: golang/167.two-sum-ii-input-array-is-sorted.go
/*
 * @lc app=leetcode id=167 lang=golang
 *
 * [167] Two Sum II - Input Array Is Sorted
 */
//  binary search is useless as it is O(nlog(n)) time complexity and O(n) space complexity
// @lc code=start
func twoSum(numbers []int, target int) []int {
	left := 0
	right := len(numbers) - 1
	sum := numbers[left] + numbers[right]
	for sum != target {
		if sum == target {
			break
		}
		if sum > target {
			right -= 1
		}
		if sum < target {
			left += 1
		}
		sum = numbers[left] + numbers[right]
	}
	return []int{left + 1, right + 1}
}

// @lc code=end

